src/corehost/cli/fxr_resolver.cpp

```diff
@@ -53,6 +53,24 @@ namespace
     }
 }
 
+bool fxr_resolver::is_runtime_embedded()
+{
+    // Check for an export of DotNetRuntime pointing to a string of 'DotNetRuntime'
+    pal::dll_t exe;
+    if (!pal::get_current_executable(&exe))
+        return false;
+
+    const char *runtime_marker = u8"DotNetRuntime";
```

Comment:
Should we call the symbol something more specific like `DotnetRuntimeEmbeddedSingleFileHost` ?

Reply:
This is the current rough purpose symbol label at this time.  If we find it isn't unique enough then it have to change.

/cc: @noahfalk

Reply:
I think we should commit to the name of the symbol -- because that check is going into 3.1 branch. 

If we have a unique enough symbol, we may not bother to check its value (so that part is free to evolve in ver 5).

Reply:
Or `EmbeddedDotNetRuntime`?
I like @swaroop-sridhar's idea of having a more unique name and just checking for its existence, such that what it actually points to is free to change in 5.

Reply:
Sounds good; Actually, we could just check one of the symbols already exported by the runtime.

Reply:
You mean a export that is in 5.0 but not 3.x or less? I'd rather be very specific with this new symbol/export.

Reply:
I would prefer checking for a very specific export instead of using some existing one as well.

Reply:
I like the idea of single-exe specific symbol - it might have other benefits (detecting single-exe-ness). And also if we're not yet signed up for a specific symbol for the diagnostics.

Reply:
I think that's reasonable. Maybe `SingleFileHost`?

Reply:
I think it should be prefixed with `Dotnet` or something along those lines. I actually like the proposal from @swaroop-sridhar - it is very specific and it is clear of what it is.

Reply:
Okay, `DotNetRuntimeEmbeddedSingleFileHost` it shall be.